### PR TITLE
Switched to Gradle for language and diagram server build

### PR DESCRIPTION
**Pull request by a-sr**
Replaces the Maven build configuration (+ Pythons script) with a Gradle task for building the LDS jar.
See [PR 85 in VSC-LF](https://github.com/lf-lang/vscode-lingua-franca/pull/85) for associated adjustments.

### CODE REVIEW

File: org.lflang.ide/build.gradle [hunk 1]
```diff
@@ -1,52 +1,7 @@
-import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 dependencies {
     implementation project(':org.lflang')
     implementation "org.eclipse.xtext:org.eclipse.xtext.ide:${xtextVersion}"
     implementation "org.eclipse.xtext:org.eclipse.xtext.xbase.ide:${xtextVersion}"
```
Comment (cmnrd) [inline org.lflang.ide/build.gradle:4]: What does this gradle file build? Is it the language server?

**Review by cmnrd — COMMENTED** [review]
Very cool! Unfortunately our CI seems to be broken at the moment. We have multiple open PRs where the workflows did not trigger.

**Comment by cmnrd** [comment]
The CI configuration was broken on master... I fixed it now.

File: org.lflang.diagram/src/org/lflang/diagram/synthesis/util/ReactorIcons.java [hunk 2]
```diff
@@ -83,68 +83,73 @@ public void handleIcon(KContainerRendering rendering, ReactorDecl reactor, boole
...
-                            _kRenderingExtensions.H_CENTRAL, _kRenderingExtensions.V_CENTRAL, 0,
-                            0, data.width, data.height);
-                }
-                if (error != null) {
-                    var errorText = _kContainerRenderingExtensions.addText(rendering, "Icon not found!\n"+error);
-                    _kRenderingExtensions.setForeground(errorText, Colors.RED);
-                    _kRenderingExtensions.setFontBold(errorText, true);
-                    _kRenderingExtensions.setSurroundingSpaceGrid(errorText, 8, 0);
-                }
+                /*
```
[the remark this answers is not in the document]
Comment (a-sr) [reply org.lflang.diagram/src/org/lflang/diagram/synthesis/util/ReactorIcons.java:86]: Good point. I created an issue and added the link in the comment.

File: org.lflang.ide/build.gradle (hunk 1 above)
Comment (a-sr) [reply org.lflang.ide/build.gradle:4]: Yes, `ide` contains the code for the xtext-generated language server. Therefore, the `diagram` plugin depends on it.
Anyway, it is only temporary as we will merge all those folders in the next restructuring step.

File: org.lflang.diagram/src/org/lflang/diagram/lsp/LanguageDiagramServer.java [hunk 3]
```diff
@@ -30,10 +36,14 @@ private static class LFLsCreator extends AbstractLsCreator {
 
         @Override
         public Module createLSModules(boolean socket) {
-            return Modules.override(super.createLSModules(socket)).with(new AbstractGenericModule() {
+            return Modules2.mixin(Modules.override(super.createLSModules(socket)).with(new AbstractGenericModule() {
                 public Class<? extends LanguageServerImpl> bindLanguageServerImpl() {
                     return LFLanguageServer.class;
                 }
+            }), it -> {
+                // Temporary fix for an issue of Klighd with Xtext 2.28 (https://github.com/kieler/KLighD/issues/144)
```
Comment (lhstrh) [inline org.lflang.diagram/src/org/lflang/diagram/lsp/LanguageDiagramServer.java:44]: We should probably create and issue to track this.

File: org.lflang.ide/build.gradle [hunk 4]
```diff
@@ -1,52 +1,7 @@
...
-    //minimize() // Minimizing might cause problems because it could delete classes that are not referenced but loaded by name (service loader)
-    from(project.convention.getPlugin(JavaPluginConvention).sourceSets.main.output)
-    configurations = [project.configurations.runtime]
-    exclude('META-INF/INDEX.LIST', 'META-INF/*.SF', 'META-INF/*.DSA', 'META-INF/*.RSA','schema/*',
-        '.options', '.api_description', '*.profile', '*.html', 'about.*', 'about_files/*',
-        'plugin.xml', 'modeling32.png', 'systembundle.properties', 'profile.list')
-    classifier = 'ls'
-    append('plugin.properties')
-}
+apply plugin: 'application'
```
Comment (lhstrh) [inline org.lflang.ide/build.gradle:7]:
```suggestion
apply plugin: 'application'

```

**Review by lhstrh — APPROVED** [review]
This looks great! Real progress :rocket:

File: org.lflang.diagram/src/org/lflang/diagram/lsp/LanguageDiagramServer.java (hunk 3 above)
Comment (a-sr) [reply org.lflang.diagram/src/org/lflang/diagram/lsp/LanguageDiagramServer.java:44]: Done. #1473

**Review by lhstrh — APPROVED** [review]

**Pull request merged**